packages/eslint-plugin-eds/src/rules/__tests__/no-h-tags.spec.ts

```diff
@@ -0,0 +1,80 @@
+import {RuleTester} from 'eslint';
+import rule from '../no-h-tags';
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+    ecmaFeatures: {jsx: true},
+  },
+});
+
+ruleTester.run('no-h-tags', rule, {
+  valid: [
+    '<Heading size="h1">Some Heading</Heading>',
+    '<Heading as="h3" size="h2">Some Heading</Heading>',
+  ],
```

Comment:
I have a complicated example that we may want to address. What are your thoughts? If we recommend this approach, I'm concerned doing so will generate lint errors

![image](https://user-images.githubusercontent.com/94476616/216731204-5b05e878-6e1a-4954-aeaa-90f8a6b37fb7.png)

source: https://eds.czi.design/0843bc428/p/8411dc-typography/b/029350

Reply:
Hmm good callout
I'm thinking I can change that docstring to be:

```tsx
<Text as="div">
  <Heading size="h1">Heading for the text passage</Heading>
  <Text>First paragraph copy of the text passage</Text>
  <Text>Second paragraph copy of the text passage</Text>
</Text>
```

What do you think?

Reply:
Do we know who originally authored the guidance and why? I'm not sure what reason this was initially suggested

Reply:
So the `as="div"` variant for the `<Text>` component is for instances when a bunch of paragraphs are written together to make a passage, like a wikipedia article. It adds vertical spacing, to space the paragraphs, or heading, blockquote, etc. apart, for each child element, depending on the element

Reply:
It might be difficult to check for the parent, if a component is a `<p>` element meant to be wrapped by a parent component `<Text as="div">`. Don't think a linter would be aware of a separate component like that?

Think the API we should recommend is using the `<Text>` tag all together even when wrapped by a `<Text as="div">`?

Reply:
I'm not sure I understand the utility of having nested `Text` elements like that. If there is some need to group related texts, it seems that we would want to use some type of un-styled container for that case. Also nesting `<p>` is not good, and nesting `<span>` is a little suspect too. 

That example needs an update, but I would think the suggestion would be an outer `div`/`section` and the contents would use `Header` and/or `Text` as needed.

Reply:
The `as="div"` option for the Text component will be deprecated https://github.com/chanzuckerberg/edu-design-system/pull/1488, so we won't be using the Text component as a container anymore. This results in no more need to increase the complexity of these lint rules